### PR TITLE
update GKE config for private gke setup

### PR DESCRIPTION
**Pull request by kunmingg**
**Which issue is resolved by this Pull Request:**

* Update gke config for private cluster setup
* Provide kfdef entry for private GKE

**Description of your changes:**

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate-changed-only`
    3. `make test`

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/990)

### CODE REVIEW

File: kfdef/kfdef_gcp_private_dex.yaml [hunk 1]
```diff
@@ -0,0 +1,306 @@
+# Kfdef config to install Kubeflow with private GKE cluster.
```
Comment (zhenghuiwang) [inline kfdef/kfdef_gcp_private_dex.yaml]: Should we follow the pattern in https://github.com/kubeflow/manifests/tree/master/kfdef to have a `kfdef_gcp_private_dex.v1.0.0.yaml`?

Same for the private_minimum.yaml

Comment (kunmingg) [reply kfdef/kfdef_gcp_private_dex.yaml]: How about publishing `v1.0.0` in separate PR, when the feature is more stable?

Comment (zhenghuiwang) [reply kfdef/kfdef_gcp_private_dex.yaml]: > How about publishing v1.0.0 in separate PR

SG.

> when the feature is more stable

Do we expect more changes to this file or we just need to test it?

Comment (kunmingg) [reply kfdef/kfdef_gcp_private_dex.yaml]: Might need more change around configs

**Comment by jlewi** [comment]
Why continue to invest in Deployment Manager as opposed to switching to KCC?

**Comment by abhi-g** [comment]
My only concern is with stability of the deployment process. With 1.0 we'd
like to ensure that the users don't have to continuously feel like things
underneath are moving so fast that couldn't keep pace. Also, upgrades will
need to ensure that they continue to work through these changes.

**Comment by jlewi** [comment]
@abhi-g What is the consistency you are going for and how does using DM help you create a consistent experience? Are you planning on switching to ASM?

**Comment by jlewi** [comment]
Closing in favor of #1126

**Pull request closed without merging**